workloads.go

```diff
@@ -15,22 +15,38 @@ func MinInt64(a int64, b int64) int64 {
 	}
 }
 
+const minToken int64 = -(1 << 63)
+const maxToken int64 = (1 << 63) - 1
+
+// Bounds are inclusive
+type TokenRange struct {
+	Start int64
+	End int64
+}
+
 type WorkloadGenerator interface {
+	NextTokenRange() TokenRange
```

Comment:
Maybe extract this to another interface `TokenRanger` and remove the dummy implementations. Also request in DoScanTable would benefit from not having to bind anything.

Reply:
After reading anther commit Restart also belongs to this interface.

Reply:
Do you suggest that only `RangeScan` workload implements this new `TokenRanger` interface? `GetMode` in main requires that all modes take a `WorkloadGenerator` interface. How can I pass in a different object just for this one case?

Reply:
`Restart` is implemented by two workloads, `Sequential` and `RangeScan`, it is not tied to the range-scan implementation.

Reply:
> Restart is implemented by two workloads, Sequential and RangeScan, it is not tied to the range-scan implementation.

My bad. Miss read.

My main point is that different workloads may decorate request / query differently there is no need to force every workload to work with token ranges. There are few approaches to that. The simplest to my understanding would be a type switch in DoScanTable. If WorkloadGenerator implements TokenRanger add the where clause to the statement and bind query. Another option could be adding GetQuery and BindQuery functions to WorkloadGenerator interface then NextTokenRange could be completely eliminated.

Reply:
For this to make sense `DoScanTable` should support more than one workloads. Currently its hard-tied to `RangeScan`, so much so that when `-mode=scan` any attempt to set the `-workload` will result in an error. This is because conceptually range scans don't support different workloads.
I agree the current code is not elegant, `WorkloadGenerator` is a union of the requirements of all modes, some of which don't make sense for other at all.

Reply:
Maybe instead of the mode is being a stateless function, we should make it stateful. Then each mode could accept only a compatible `WorkloadGenerator` when constructed. The inner execution loop would then store the mode object (through some minimal interface), not the `WorkloadGenerator`. This we could have tailored workloads for each mode, or share some workloads between modes, but only as far as it makes sense.

Reply:
I guess this is a valid idea but deserves a separate PR. To minimize work how about `DoScanTable` accepted `*RangeScan` and `NextTokenRange()` was removed from the interface?

Reply:
That is doable, `RunConcurrently` doesn't know about the mode or the workload type.

Reply:
:+1:

Reply:
I had a look at this and I'm afraid It can't be done without extensive refactoring. `RunTest()`, which is the inner loop actually running the tests knows about the `WorkloadGenerator` type, so separating this would expose concrete implementations to it.